File: src/test/java/it/xpug/kata/birthday_greetings/XDateTest.kt
package it.xpug.kata.birthday_greetings

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class XDateTest {
    @Test
    fun getters() {
        val date = XDate("1789/01/24")

        assertThat(date.month).isEqualTo(1)
        assertThat(date.day).isEqualTo(24)
    }

    @Test
    fun isSameDate() {
        val date = XDate("1789/01/24")
        val sameDay = XDate("2001/01/24")
        val notSameDay = XDate("1789/01/25")
        val notSameMonth = XDate("1789/02/25")

        assertThat(date.isSameDay(sameDay))
            .`as`("same")
            .isTrue()
        assertThat(date.isSameDay(notSameDay))
            .`as`("not same day")
            .isFalse()
        assertThat(date.isSameDay(notSameMonth))
            .`as`("not same month")
            .isFalse()
    }
}

File: csharp/BirthdayGreetings.Tests/XDateTest.cs
using NUnit.Framework;
using BirthdayGreetings;

namespace BirthdayGreetings.Tests
{
    [TestFixture]
    public class XDateTest
    {
        [Test]
        public void Getters()
        {
            var date = new XDate("1789/01/24");
            Assert.That(date.Month, Is.EqualTo(1));
            Assert.That(date.Day, Is.EqualTo(24));
        }

        [Test]
        public void IsSameDate()
        {
            var date = new XDate("1789/01/24");
            var sameDay = new XDate("2001/01/24");
            var notSameDay = new XDate("1789/01/25");
            var notSameMonth = new XDate("1789/02/25");

            Assert.That(date.IsSameDay(sameDay), Is.True, "same");
            Assert.That(date.IsSameDay(notSameDay), Is.False, "not same day");
            Assert.That(date.IsSameDay(notSameMonth), Is.False, "not same month");
        }
    }
} 